snuba/datasets/tags_column_processor.py

```diff
@@ -54,12 +104,20 @@ def process_column_expression(
 
         It returns None if the column is not a tag or context.
         """
-        matched = NESTED_COL_EXPR_RE.match(column_name)
-        if matched and matched[1] in ["tags", "contexts"]:
-            return self.__tag_expr(column_name, table_alias)
-        elif column_name in ["tags_key", "tags_value"]:
-            return self.__tags_expr(column_name, query, parsing_context, table_alias)
-        return None
+        parsed_col = ParsedNestedColumn.parse_column_expression(column_name)
+        if not parsed_col:
+            return None
+        if parsed_col.is_single_context() or parsed_col.is_single_tag():
+            return self.__tag_expr(parsed_col, table_alias)
+        elif (
+            parsed_col.is_all_tags_keys()
+            or parsed_col.is_all_tags_values()
+            or parsed_col.is_selected_tags_keys()
+        ):
+            # TODO: Should we support contexts?
```

Comment:
to be clear, contexts are were not supported before this PR either.

Reply:
I can't think of a reason we'd want to add this (except maybe consistency). Seems like quite a bit of extra work to support queries which expand both tags and contexts.